### PR TITLE
Pfmuon fix 90x

### PR DESCRIPTION
**Pull request by jshlee**
low tight and loose muon ID efficiency due to low pfmuon efficiency due to GeneralTracksImporterWithVeto killing pftracks
This issue was noticed in #16797 
pfmuon temp fix - hgcal will fix later 

fix rotation in me0geometry - in me0 even and odd chambers are not inverted

@calabria @lgray @pietverwilligen @kpedro88

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by kpedro88** [comment]
+1

**Comment by ianna** [comment]
+1

File: Geometry/GEMGeometryBuilder/src/ME0GeometryBuilderFromDDD10EtaPart.cc [hunk 1]
```diff
@@ -366,8 +366,7 @@ ME0GeometryBuilderFromDDD10EtaPart::boundPlane(const DDFilteredView& fv,
   Basic3DVector<float> newX(1.,0.,0.);
   Basic3DVector<float> newY(0.,0.,1.);
   Basic3DVector<float> newZ(0.,1.,0.);
-  // Odd chambers are inverted in gem.xml
-  if (isOddChamber) newY *= -1;
+  newY *= -1;
```
Comment (slava77) [inline Geometry/GEMGeometryBuilder/src/ME0GeometryBuilderFromDDD10EtaPart.cc]: this does not belong in the same PR, please split.

**Comment by slava77** [comment]
-1

I think this should be split in two (ME0 geometry removed from this PR)

**Comment by jshlee** [comment]
@slava77 - done!
will open a new one for me0

**Comment by slava77** [comment]
@cmsbuild please test

@jshlee thank you for the quick update

**Comment by slava77** [comment]
@cmsbuild please test
restart after the frontier storm is over

**Comment by slava77** [comment]
@jshlee 
please provide some more details on the upcoming fixes and also on expected changes from this PR alone.

**Comment by jshlee** [comment]
@slava77 - there are no upcoming fixes from the muon side. hgcal will fix this later i guess.

**Comment by slava77** [comment]
@jshlee 
this PR changes physics outputs (pf muons, jets etc),
based on jenkins tests (no pileup and only ~10 events, which doesn't help to understand changes in a significant manner).
I was asking for details of what changes are expected with some higher stat sample and comparisons.

**Comment by jshlee** [comment]
@slava77 - sorry i missunderstood. 
Details can be found in this talk
https://indico.cern.ch/event/614328/contributions/2477239/attachments/1414387/2164782/9_0_0_pre4_results__isPFMuon_issue_-_fixed.pdf
This slide shows a lot of muons within eta 1.5 to 2.4 are not made into pfmuons, which makes the loose and tight identification selections fail.
screen shot 2017-03-07 at 10 18 57 pm

**Comment by slava77** [comment]
assign upgrade

I suppose Kevin's "+1" will be accepted retroactively.

**Comment by kpedro88** [comment]
+1
(though this only needs to go in 91X at this point)

**Comment by slava77** [comment]
Looking at PF particles distributions, it seems to me there are more things broken now than fixed.

In TenMuExtended wf 21211
all_sign873vsorig_tenmuextendede2023d4wf21211p0c_recopfcandidates_particleflow__reco_obj_eta43

all_sign873vsorig_tenmuextendede2023d4wf21211p0c_recopfcandidates_particleflow__reco_obj_eta57

it looks like muons are double-counted as charged hadrons in eta range 2.5 to 3.
Also, muons themselves appear to be double-counted in HGCAL region (the baseline flattish distribution is more what I would expect from a flat spectrum GEN)

In ttbar PU200 
pf charged hadron population nearly doubles in HGCAL
shouldn't it remain somewhat unchanged? all_sign873vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recopfcandidates_particleflow__reco_obj_eta43

all_sign873vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recopfcandidates_particleflow__reco_obj_eta57

MET goes further away from the true value from ~50  to ~150-200
wf23034pu200_pfmet_gendiff

**Comment by jshlee** [comment]
@slava77 - Yes, this PR did fix the problem for muons. I don't know what change...
Do you have any suggestions??
We would like the pfmuons to be fixed for the TDR MC productions.

**Comment by slava77** [comment]
On 3/9/17 10:06 PM, Jason Lee wrote:
> @slava77 <https://github.com/slava77> - Yes, this PR did fix the problem
> for muons. I don't know what change...
> Do you have any suggestions??
> We would like the pfmuons to be fixed for the TDR MC productions.

Can you clarify what is achieved by fixing this "isPFmuon" this way
compared to just ignoring the "isPFmuon" flag in the tight/loose ID?

If nothing else changed, the fix makes sense, but that's not the case
and the fix is not a consistent improvement (known already from the
linked slides even at just fake rate level).

**Comment by slava77** [comment]
this looks like a "-1" to me.

@kpedro88 
please clarify if from some arguments this change is actually a desired improvement overall

**Comment by lgray** [comment]
Hi, this cross cleaning was originally there to remove the double counting
that slava sees.

This means the effect for muons needs to be targeted and you need to use
the vetoed track list.

**Comment by kpedro88** [comment]
My +1 was only based on the demonstrated improvements for muons. I agree with @lgray, the HGCal cleaning should only be removed for muons rather than all PF candidates.

**Comment by kpedro88** [comment]
-1
(based on further tests by @slava77, pending a more muon-specific implementation of the fix)

**Comment by jshlee** [comment]
yes I agree too.
I will look into this further

**Comment by slava77** [comment]
-1

based on already observed issues.

Also, it looks like this will miss or is not really needed in 90X.
We should move to #17703 in this case.

**Pull request closed without merging**